src/views/user/ResetPassword/ResetPassword.jsx

```diff
@@ -0,0 +1,307 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { PropTypes } from 'prop-types';
+import axiosInstance from '../../../services/requestHandler';
+import resetPasswordActions from '../../../actions/resetPasswordActions';
+import ResetPasswordForm from './ResetPasswordForm';
+import InvalidTokenMessage from './InvalidTokenMessage';
+
+const { setToken, updatePassword } = resetPasswordActions;
+
+/**
+ *
+ * The ResetPassword component renders view to get
+ * user email and hits the route that sends reset
+ * password link to user email
+ * @class ResetPassword
+ * @returns {undefined} this is a class thus does not return anything
+ */
+export class ResetPassword extends React.Component {
+  state = {
+    password: '',
+    confirm: '',
+    disabled: true,
+    isValidToken: false,
+    frontendValidations: {
+      password: [],
+      confirm: [],
+    },
+  };
+
+  validity = {
+    passwordIsValid: false,
+    passwordIsConfirmed: false,
+  };
+
+  frontendValidations = {
+    password: [],
+    confirm: [],
+  };
+
+  /**
+   * componentDidMoutn
+   * @returns {null} null
+   */
+  componentDidMount() {
+    const { location, dispatch, history } = this.props;
+    const resetToken = location.search.substr(7);
+    axiosInstance().get(`/users/reset-password?token=${resetToken}`)
```

Comment:
I won't allow you to do this in the context of a react component, pls move this to an action to update state, that's what actions are for, cos if this call errors, how are we going to set an error state in redux store